packages/launch/src/hooks/use-domain-search.ts

```diff
@@ -17,13 +18,15 @@ export function useDomainSearch(): {
 } {
 	const { domainSearch } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
 	const { title } = useTitle();
-	const { currentDomainName, isLoadingSite } = useSite();
+	const { isLoadingSite } = useSite();
+	const { siteSubdomain } = useSiteDomains();
+
 	const { setDomainSearch } = useDispatch( LAUNCH_STORE );
 
 	let search = domainSearch.trim() || title;
 
 	if ( ! search || isDefaultSiteTitle( { currentSiteTitle: search, exact: true } ) ) {
-		search = currentDomainName?.split( '.' )[ 0 ] ?? '';
+		search = siteSubdomain?.domain?.split( '.' )[ 0 ] ?? '';
```

Comment:
This logic might need a bit of love to become more readable. The idea is use one of the following for domain search, in this order:
1. `domainSearch`
2. `title` if valid
3. `siteSubdomain?.domain?.split( '.' )[ 0 ]`
4. `''`

Also, it feels like `isValidSiteTitle` would be a better choice than `isDefaultSiteTitle`.

However, this isn't blocking for this PR.

Reply:
I will create a janitorial type issue for this.